sysapps/common/common_api.js

```diff
@@ -80,13 +71,18 @@ var BindingObjectPrototype = function() {
     });
   };
 
-  function addMethodWithPromise(name, Promise) {
+  function addMethodWithPromise(name) {
     Object.defineProperty(this, name, {
       value: function() {
-        var promise_instance = new Promise();
         var args = Array.prototype.slice.call(arguments);
-        this._postMessage(name, args, wrapPromiseAsCallback(promise_instance));
-        return promise_instance;
+        return new Promise((resolve, reject) => {
+          this._postMessage(name, args, function(data, error_message) {
+            if (error_message)
+              reject(new Error(error_message));
```

Comment:
Did you test that? I believe the message has to be second argument, but I might be mistaken that that is just for ErrorEvent